src/mame/drivers/sm7238.cpp

```diff
@@ -21,26 +22,27 @@
 #include "machine/bankdev.h"
 #include "machine/clock.h"
 #include "machine/i8251.h"
-#include "machine/pit8253.h"
-#include "machine/pic8259.h"
 #include "machine/km035.h"
 #include "machine/nvram.h"
+#include "machine/pic8259.h"
+#include "machine/pit8253.h"
+
 #include "emupal.h"
 #include "screen.h"
 
 
-//#define VERBOSE (LOG_DEBUG)
-//#define LOG_OUTPUT_FUNC printf
-#include "logmacro.h"
+static constexpr int KSM_COLUMNS_MAX = 132;
 
+static constexpr int KSM_TOTAL_HORZ = KSM_COLUMNS_MAX * 10;
+static constexpr int KSM_DISP_HORZ = KSM_COLUMNS_MAX * 8;
 
-#define KSM_COLUMNS_MAX 132
+static constexpr int KSM_TOTAL_VERT = 260;
+static constexpr int KSM_DISP_VERT = 250;
 
-#define KSM_TOTAL_HORZ (KSM_COLUMNS_MAX*10)
-#define KSM_DISP_HORZ  (KSM_COLUMNS_MAX*8)
 
-#define KSM_TOTAL_VERT 260
-#define KSM_DISP_VERT  250
+//#define VERBOSE (LOG_DEBUG)
+//#define LOG_OUTPUT_FUNC printf
```

Comment:
It’s better to use `//#define LOG_OUTPUT_FUNC osd_printf_info` or `//#define LOG_OUTPUT_STREAM stderr` as the example for redirecting the logging.  The C `printf` family functions have different semantics to `logerror`, the `osd_printf_*` family functions, and the `util::stream_format` function used when `LOG_OUTPUT_STREAM` is defined.

Reply:
Changed